src/utils/routes/routeCosts.ts

```diff
@@ -0,0 +1,159 @@
+      if (step.includedSteps && step.includedSteps.length > 0) {
+        step.includedSteps.forEach((includedStep) => {
+          // Process gas costs
+          if (includedStep.estimate.gasCosts) {
+            const gasAmount = includedStep.estimate.gasCosts.reduce(
+              (sum, gasCost) => sum + parseFloat(gasCost.amount || '0'),
+              0,
+            );
+            const gasAmountUSD = includedStep.estimate.gasCosts.reduce(
+              (sum, gasCost) => sum + parseFloat(gasCost.amountUSD || '0'),
+              0,
+            );
+            totalGasAmount += gasAmount;
+            totalGasAmountUSD += gasAmountUSD;
+          }
+
+          // Process fee costs
+          if (includedStep.estimate.feeCosts) {
+            const feeAmount = includedStep.estimate.feeCosts.reduce(
+              (sum, feeCost) => sum + parseFloat(feeCost.amount || '0'),
+              0,
+            );
+            const feeAmountUSD = includedStep.estimate.feeCosts.reduce(
+              (sum, feeCost) => sum + parseFloat(feeCost.amountUSD || '0'),
+              0,
+            );
+            totalFeeAmount += feeAmount;
+            totalFeeAmountUSD += feeAmountUSD;
+          }
+        });
+      }
+    });
+
+    // Get decimals from the first token in gas/fee costs for formatting
+    const includedGasToken = route.steps.find((step) =>
+      step.includedSteps?.find(
+        (included) => included.estimate.gasCosts?.[0]?.token,
+      ),
+    )?.includedSteps?.[0]?.estimate.gasCosts?.[0]?.token;
+
+    const includedFeeToken = route.steps.find((step) =>
+      step.includedSteps?.find(
+        (included) => included.estimate.feeCosts?.[0]?.token,
+      ),
+    )?.includedSteps?.[0]?.estimate.feeCosts?.[0]?.token;
+
+    const formattedIncludedGasAmount =
+      includedGasToken &&
+      formatUnits(BigInt(totalGasAmount.toString()), includedGasToken.decimals);
+    const formattedIncludedFeeAmount =
+      includedFeeToken &&
+      formatUnits(BigInt(totalFeeAmount.toString()), includedFeeToken.decimals);
+
+    return {
+      gasCostUSD: totalGasAmountUSD,
+      gasCost: totalGasAmount,
+      gasCostFormatted: formattedIncludedGasAmount,
+      feeCostUSD: totalFeeAmountUSD,
+      feeCost: totalFeeAmount,
+      feeCostFormatted: formattedIncludedFeeAmount,
+    };
+  } else if (
+    route.steps.find((step) => 'execution' in step || 'estimate' in step)
+  ) {
+    // Use regular costs logic
+    let totalGasAmount = 0;
+    let totalGasAmountUSD = 0;
+    let totalFeeAmount = 0;
+    let totalFeeAmountUSD = 0;
+
+    route.steps.forEach((step) => {
+      // Check if the step is of type LiFiStepExtended by checking the presence of the 'execution' property
+      const isExtendedStep = 'execution' in step;
+      // Determine detailInformation: use 'execution' if it's an extended step, otherwise fall back to 'estimate'
+      const detailInformation =
+        isExtendedStep && step.execution
```

Comment:
Duplication of https://github.com/jumperexchange/jumper-exchange/pull/2005/files#diff-0d5c02ba42e587115fdc3ea43411e9d78e54b19acb368e8fc297249bdb38e462R28